src/os_posix/os_mtx_rw.c

```diff
@@ -159,22 +156,17 @@ __wt_try_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *rwlock)
 	 * that is, if this thread's ticket would be the next ticket granted.
 	 * Do the cheap check to see if this can possibly succeed.
 	 */
-	if (l->s.readers != users)
+	if (old.s.readers != old.s.users)
 		return (EBUSY);
 
 	/*
 	 * The old and new values for the lock: note the use of "users" instead
-	 * of "writers", this may not be the lock's current value, rather it's
-	 * the value the lock must have if we are to grant this write lock.
-	 *
-	 * Note the masking of "users + 1" in the calculation of the new value:
-	 * we want to set the readers field to the next readers value, not the
-	 * next users value, so it has to wrap, not overflow.
+	 * of "readers", this may not be the lock's current value, rather it's
+	 * the value the lock must have if we are to grant this read lock.
```

Comment:
I think we should make this comment change regardless of the code changes: there was a copy/paste swap of "write" for "read" that I didn't spot on my review.